openedx/core/djangoapps/discussions/views.py

```diff
@@ -61,4 +122,121 @@ def post(self, request, course_key_string: str, **_kwargs) -> Response:
                 check_course_permissions(course, request.user, 'change_provider')
 
             serializer.save()
-        return Response(serializer.data)
+        return serializer.data
+
+
+class DiscussionsProvidersView(APIView):
+    """
+    Read only view that lists details of discussion providers available for a course.
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser
+    )
+    permission_classes = (IsStaffOrCourseTeam,)
+
+    @apidocs.schema(
+        parameters=[
+            apidocs.string_parameter(
+                'course_id',
+                apidocs.ParameterLocation.PATH,
+                description="The course for which to get provider list",
+            )
+        ],
+        responses={
+            200: DiscussionsProvidersSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    def get(self, request, course_key_string: str, **_kwargs) -> Response:
+        """
+        Handle HTTP/GET requests
+        """
+        data = self.get_provider_data(course_key_string)
+        return Response(data)
+
+    @staticmethod
+    def get_provider_data(course_key_string: str) -> Dict:
+        """
+        Get provider data for specified course
+        Args:
+            course_key_string (str): course key string
+
+        Returns:
+            Dict: course discussion providers
+        """
+        course_key = validate_course_key(course_key_string)
+        configuration = DiscussionsConfiguration.get(course_key)
+        hidden_providers = []
+        # If the user is currently using the legacy provider, don't show the new provider
+        # TODO: Allow switching between legacy and new providers
+        if configuration.provider_type == Provider.LEGACY:
+            hidden_providers.append(Provider.OPEN_EDX)
```

Comment:
Can you explain if the course team has selected a legacy provider, why we are hiding open-edx provider for them, and vice versa?  

How a course can select open_edx provider if the default provider every course get is "legacy"?

Reply:
Right now, they can't, which is kind of by design since the new provider isn't ready, and we don't have a way to migrate between them. When there is a migration path etc, we can look to provide a way to switch over.